### ISSUE TITLE
Reading the Shop Page ID

### ISSUE THREAD

**Issue by skaparate** [gocodebox/lifterlms#1247]
Hello,

I'm not sure if this is a bug or not, that's why I post it as a question.

In the file [includes/class.llms.template.loader.php](https://github.com/gocodebox/lifterlms/blob/d03d07630f827cdab3f725920393a54405224397/includes/class.llms.template.loader.php#L465), line 465, there is this check:

```php
elseif ( is_post_type_archive( 'course' ) || is_page( llms_get_page_id( 'llms_shop' ) ) ) {
```

`llms_shop` appears to be an error, since it's not created anywhere on the (base?) code (I've searched for it and could not find it), specially since the `llms_get_page_id` doesn't seem to recognize the page, unless it's used by another plugin that I'm not aware of:

```php
function llms_get_page_id( $page ) {
	// Normalize some pages to make more sense without having to migrate options.
	if ( 'courses' === $page ) {
		$page = 'shop';
	}

	$id = get_option( 'lifterlms_' . $page . '_page_id' );

	/**
	 * Filter the ID of the requested LifterLMS Page
	 *
	 * The dynamic portion of this filter, {$page}, refers to the LifterLMS page slug/name.
	 *
	 * Note that, historically, the course catalog was called the "shop" and therefore when requesting
	 * the filter will be "lifterlms_get_shop_page_id" instead of "lifterlms_get_courses_page_id".
	 *
	 * @since 1.0.0
	 *
	 * @param int|string $id The WP_Post ID of the requested page or an empty string if the page doesn't exist.
	 */
	$page = apply_filters( "lifterlms_get_{$page}_page_id", $id );

	return $page ? absint( $page ) : -1;

}
```

Thanks for your attention,

Regards!

**Comment by thomasplevy** [comment]
@skaparate 

Thanks for your bug report. At a glance this does look like a bug, however, I'm struggling to find the actual impact of this bug as it looks like the condition itself may be superfluous. Can you help me with some reproduction steps that illustrate the impact of this issue?

Thanks

**Comment by thomasplevy** [comment]
@skaparate thanks for your bug report, this does look like a typo -- it should be `shop` not `llms_shop`, however, I don't see any way to actually create an issue as a result of this bug. The or condition here appears to be more or less superfluous (another problem). I might be missing something though. 

Do you have some recreation steps you can share that illustrate the impact of the bug so we can use that to clean this up and fix it.


Thanks,

**Comment by skaparate** [comment]
@thomasplevy Thanks for your answer!

I'm not really sure if it's a bug xD (I selected the question category when I created the topic, but I must have removed it later)...

The thing is, I was trying to set the page to the course catalog to a different page (nuestros-cursos in spanish), but that didn't work. Note that I'm not trying to modify the page in any way, **it's a basic page with the title set as the catalog page**, but it doesn't allow me. It only works if the **slug** is *courses*.

However, if I change **llms_shop** to shop, it works. I can work with the default **slug** (courses), I just wanted to let you know and be sure that this is the expected behavior.

The steps to reproduce:

1. Create a new page and set a title, assuming the default one is created.
2. Change the page settings to set the catalog page to this newly created page.
3. Navigate to this page (on the front end) and should display nothing. Whereas if you go to the default page (/courses, assuming you don't remove it) the courses are shown normally, but with the title of the new page.

Here are some pictures:

### Default Catalog Page

![image](https://user-images.githubusercontent.com/12261705/88430564-a7c2c680-cdc6-11ea-8b61-c772e429c7fd.png)

### Default Lifter Courses Page Settings

![image](https://user-images.githubusercontent.com/12261705/88430657-d17bed80-cdc6-11ea-8f3f-7832940eb66e.png)

### Default Catalog Page, Front End

![image](https://user-images.githubusercontent.com/12261705/88430745-f7a18d80-cdc6-11ea-9780-7d36e18243e1.png)

### Create New Page

![image](https://user-images.githubusercontent.com/12261705/88430794-09833080-cdc7-11ea-855f-2257ef4391a1.png)

### Updated Lifter Settings

As you can see, I've updated the settings to use the newly created page:

![image](https://user-images.githubusercontent.com/12261705/88430892-359eb180-cdc7-11ea-8244-7c6bdb09538d.png)

### New Course Catalog Page

Notice the slug on the URL, in contrast to the one on the Default Catalog Page.

![image](https://user-images.githubusercontent.com/12261705/88431031-86aea580-cdc7-11ea-85d2-f83b921cd9f7.png)

### Default Course Catalog, Updated Title

Notice how the only change is the page title:

![image](https://user-images.githubusercontent.com/12261705/88431015-7d253d80-cdc7-11ea-96ae-becbb82b6703.png)

Thanks for your time :),

Regards!

**Comment by thomasplevy** [comment]
@skaparate this has fallen off my radar and I'm sorry about that, this will be addressed in our next release, see #1455 if you're interested.

Thanks for your patience!
